### PR TITLE
[RFC] oldtest: set helpfile for runtime/doc/tags in build dir

### PR DESCRIPTION
**Pull request by blueyed**
This is better practice in general, and allows to remove the "helptags
ALL" hacks.

Ref: https://github.com/neovim/neovim/issues/8824
Ref: https://github.com/neovim/neovim/commit/f1b67c3453c

### CODE REVIEW

**Comment by blueyed** [comment]
Ok, that would need to copy/configure the runtime, which is a bit too much I guess.

The problem here is that Vim ships `runtime/doc/tags`, while Neovim does not - it seems to be forgotten in a98a6996c maybe (at least it is not explicitly mentioned).

**Comment by justinmk** [comment]
> The problem here is that Vim ships `runtime/doc/tags`, while Neovim does not - it seems to be forgotten in [a98a699](https://github.com/neovim/neovim/commit/a98a6996c291b3e300d27b791eded9eed333d677) maybe (at least it is not explicitly mentioned).

wasn't forgotten, we intentionally decided not to include generated files in the tree. Doing that isn't much better of a "hack" than simply running `:helptags ALL` ...

**Comment by blueyed** [comment]
Ok.  I can see the reason for it.

I think using `set helpfile=…` is a better/smaller diff then to get Vim's behavior in tests.

**Comment by justinmk** [comment]
> I think using `set helpfile=…` is a better/smaller diff then to get Vim's behavior in tests.

That changes the assumed state though. Should be done in specific tests maybe.

**Comment by blueyed** [comment]
> That changes the assumed state though.

The state is different however.

What about adding the runtime from the build dir?  (latest commit)

**Comment by blueyed** [comment]
Should also use `make oltest` on AppVeyor then - but having it built via functionaltest before might be good.  Code ref: https://github.com/neovim/neovim/blob/46cb85691c18f99af70de65d0168e972b1dc57dd/ci/build.ps1#L137-L143

**Comment by blueyed** [comment]
There are failures with adding build/runtime to rtp always: https://travis-ci.org/neovim/neovim/jobs/574180741#L6426

What do you think?
Reverted 384e519c7 for now.

File: test/functional/autocmd/bufenter_spec.lua [hunk 1]
```diff
@@ -33,9 +33,10 @@ describe('autocmd BufEnter', function()
   end)
 
   it('triggered by ":split normal|:help|:bw"', function()
+    -- Add runtime from build dir for doc/tags (used with :help).
+    command(string.format([[set rtp+=%s/runtime]], helpers.test_build_dir))
     command("split normal")
     command("wincmd j")
-    command("helptags runtime/doc")
     command("help")
```
Comment (blueyed) [inline test/functional/autocmd/bufenter_spec.lua]: Maybe we could have a `TestHelp` command instead that would adjust `rtp` then before calling `help`?
But it's also OK maybe having it done 5 times manually?

Comment (justinmk) [reply test/functional/autocmd/bufenter_spec.lua]: a function would be useful, if there's some reason we can't just do it once in `clear()`.

Comment (blueyed) [reply test/functional/autocmd/bufenter_spec.lua]: https://github.com/neovim/neovim/commit/384e519c78249fe59c1e3945828bd8d276d2ba3b did not work - i.e. that would need other test adjustments then.

File: src/nvim/testdir/test_taglist.vim [hunk 2]
```diff
@@ -89,8 +89,11 @@ func Test_tagfiles()
 
   help
   let tf = tagfiles()
+  " Nvim: expectation(s) based on tags in build dir (added to &rtp).
+  "       Filter out the (non-existing) '../../../runtime/doc/tags'.
+  call filter(tf, 'filereadable(v:val)')
```
Comment (justinmk) [inline src/nvim/testdir/test_taglist.vim]: is this still needed? why not do `helptags ALL` once?

File: CMakeLists.txt [hunk 3]
```diff
@@ -541,6 +541,7 @@ add_subdirectory(test/includes)
 add_subdirectory(config)
 add_subdirectory(test/functional/fixtures)  # compile test programs
 add_subdirectory(runtime)
+get_directory_property(GENERATED_HELP_TAGS DIRECTORY runtime DEFINITION GENERATED_HELP_TAGS)
```
Comment (justinmk) [inline CMakeLists.txt]: is this redefining `GENERATED_HELP_TAGS`? should it use a different var name, e.g. `GENERATED_HELP_TAGS_DIR`

File: test/functional/autocmd/bufenter_spec.lua (hunk 1 above)
Comment (justinmk) [reply test/functional/autocmd/bufenter_spec.lua]: No, I mean just do whatever you're already doing, except in `clear()`.

File: CMakeLists.txt (hunk 3 above)
Comment (blueyed) [reply CMakeLists.txt]: It is pulling it in from the subdirectory.
Could have a different name, but it is the same after all.

File: test/functional/autocmd/bufenter_spec.lua (hunk 1 above)
Comment (blueyed) [reply test/functional/autocmd/bufenter_spec.lua]: So the nearest `clear()` to the tests that need it, or in the default/global clear?  (the latter would be the same I guess with regard to unrelated/different failures then)

Comment (justinmk) [reply test/functional/autocmd/bufenter_spec.lua]: I was thinking global clear, but sounds like that is too much trouble. It's better to explicit I guess, anyway (i.e. `helpers.add_builddir_to_rtp()`).

Comment (blueyed) [reply test/functional/autocmd/bufenter_spec.lua]: Good idea, done.

**Review by justinmk — APPROVED** [review]

**Pull request merged**